FRs #1055/1073: Refactor Provider Identification, Host Provider, Invoice Issuer (#1139)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/datasets/cost_and_usage/columns/serviceprovidername.md

# Service Provider Name

Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovider), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios. In marketplace scenarios, the Service Provider is not the entity operating the marketplace, as marketplace is providing a mechanism for purchase, not independently providing the resources or services independently for purchase or usage.

The Service Provider Name column adheres to the following requirements:

* The Service Provider column MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset).

> * This column MUST be of type String and MUST NOT contain null values.

Reviewer: Let's use precisely the normative requirements as they were stated in `ProviderName`, which follow the normative requirements guidelines.

```suggestion
* ServiceProviderName MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset).
* ServiceProviderName MUST be of type String.
* ServiceProviderName MUST conform to [StringHandling](#stringhandling) requirements.
* ServiceProviderName MUST NOT be null.
```

# Service Provider Name

> Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovider), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios. In marketplace scenarios, the Service Provider is not the entity operating the marketplace, as marketplace is providing a mechanism for purchase, not independently providing the resources or services independently for purchase or usage.

Reviewer 2: Struggling with the first sentence 
> Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. 

To me i could easily see when snowflake is made available on the AWS marketplace, someone could read that as 'AWS'  was the entity that made that service available for 'purchase'.

if we instead, remove the 'for purchase' part, it becomes a little clearer:
Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage.

Reviewer 3: or even "Service Provider Name is the name of the entity that provides the resources or services."

Author: ```suggestion
Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovider), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios. In marketplace scenarios, the Service Provider is the seller, not the entity operating the marketplace, as marketplace is providing a mechanism for purchase, not independently providing the resources or services independently for purchase or usage.
```

> Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovider), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios. In marketplace scenarios, the Service Provider is the seller, not the entity operating the marketplace, as marketplace is providing a mechanism for purchase, not independently providing the resources or services independently for purchase or usage.

Reviewer 3: ```suggestion
Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovidername), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios. 

**Note:** In marketplace scenarios, the Service Provider represents the seller rather than the marketplace operator, as the marketplace operator merely provides a purchasing mechanism and does not itself provide the *resources* or *services* being consumed.
```

The Service Provider Name column adheres to the following requirements:

* ServiceProviderName MUST be present in a [*FOCUS dataset*](#glossary:FOCUS-dataset).
* ServiceProviderName MUST be of type String.
* ServiceProviderName MUST conform to [StringHandling](#stringhandling) requirements.
* ServiceProviderName MUST NOT be null.

> See [Appendix: Entity Identification Examples](#entityidentification) section for examples of Service Provider values for various use case scenarios.

Reviewer: ```suggestion
See [Appendix: Entity Identification Examples](#entityidentification) section for examples of Service Provider Name values for various use case scenarios.
```

> See [Appendix: Entity Identification Examples](#entityidentification) section for examples of Service Provider Name values for various use case scenarios.

Reviewer 3: ```suggestion
See [Appendix: Participating Entity Identification Examples](#participatingentityidentificationexamples) section for examples of Service Provider Name values across various use case scenarios.
```

## Column ID

ServiceProviderName

## Display Name

Service Provider Name

## Description

The name of the entity that made the *resources* or *services* available for purchase or consumption.

## Content Constraints

| Constraint      | Value           |
|:----------------|:----------------|
| Column type     | Dimension       |
| Feature level   | Mandatory       |
| Allows nulls    | False           |
| Data type       | String          |
| Value format    | \<not specified> |

## Introduced (version)

> 1.3 Introduced as a replacement for ProviderName

Reviewer: ```suggestion
1.3 Introduced as a replacement for [ProviderName](#providername)
```

# Service Provider Name

Service Provider Name is the name of the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovidername), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios.

> **Note:** In marketplace scenarios, the Service Provider represents the seller rather than the marketplace operator, as the marketplace operator merely provides a purchasing mechanism and does not itself provide the *resources* or *services* being consumed.

Reviewer: @rileyjenk to add reseller scenario to this note, or another note, to provide clarity as to what belongs here.

Author: This is now done.

# Service Provider

> A Service Provider is the entity that provides the  [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovider), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios. In marketplace scenarios, the Service Provider is not the entity operating the marketplace, as marketplace is providing a mechanism for purchase, not independently providing the resources or services independently for purchase or usage.

Reviewer 4: nit
```suggestion
A Service Provider is the entity that provides the [*resources*](#glossary:resource) or [*services*](#glossary:service) available for usage or purchase. These services can be built on top of infrastructure provided by a [Host Provider](#hostprovider), offered as fully integrated solutions, or include complementary offerings such as support, licensing, or consulting. It is commonly used for cost analysis and reporting scenarios. In marketplace scenarios, the Service Provider is not the entity operating the marketplace, as marketplace is providing a mechanism for purchase, not independently providing the resources or services independently for purchase or usage.
```
